services/core/shared/core/compat/sdk_v5/blocksUtils.js

```diff
@@ -23,25 +24,43 @@ const {
 	HTTP: { request },
 } = require('lisk-service-framework');
 
+const { getApiClient } = require('../common/wsRequest');
 const config = require('../../../../config');
 
 const logger = Logger();
 
-const genesisBlockURL = config.endpoints.genesisBlock;
+let readStream;
+let genesisBlockURL;
+let genesisBlockFilePath;
+let genesisBlock = { header: {} };
 
-const genesisBlockFilePath = './shared/core/compat/sdk_v5/static/genesis_block.json';
+const parseStream = json.createParseStream();
 
-let genesisBlockId;
+const setGenesisBlock = (block) => genesisBlock = block;
 
-const setGenesisBlockId = (id) => genesisBlockId = id;
+const getGenesisBlock = () => genesisBlock;
 
-const getGenesisBlockId = () => genesisBlockId;
+const getGenesisBlockId = () => genesisBlock.header.id;
+
+const loadConfig = async () => {
+	// Direct invocation of action necessary to avoid circular dependency
+	const apiClient = await getApiClient();
+	const { networkIdentifier } = await apiClient.node.getNodeInfo();
+
+	const [networkConfig] = config.network.filter(c => c.identifier === networkIdentifier);
+	genesisBlockURL = networkConfig.genesisBlockUrl;
+
+	genesisBlockFilePath = `./shared/core/compat/sdk_v5/static/${networkConfig.name}/genesis_block.json`;
```

Comment:
Please store that files in a separate dir `./data/${networkConfig.name}`. Please add the data path to gitignore and dockerignore.